### PR TITLE
ppc64le build support

### PR DESCRIPTION
**Pull request by Sapana-Khemkar**
This PR supports build for ppc64le.
We have completed testing of library tests. All tests passed on v6.0.2. Few test cases failed on main branch. But those are not blockers.
PAL layer changes are in progress and coming soon.

Note: This is breakup for PR [67378](https://github.com/dotnet/runtime/pull/67378 ). PR has been broken up in multiple PRs for easy review.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
@dotnet/area-meta do we need an `arch-ppc64le` label?

**Comment by directhex** [comment]
@teo-tsirpanis yes.

**Review by directhex — APPROVED** [review]

File: eng/native/build-commons.sh [hunk 1]
```diff
@@ -392,6 +392,10 @@ while :; do
             __TargetArch=wasm
             ;;
 
+	ppc64le|-ppc64le)
```
Comment (akoeplinger) [inline eng/native/build-commons.sh:395]: spaces for indentation:
```suggestion
        ppc64le|-ppc64le)
```

File: eng/native/configureplatform.cmake [hunk 2]
```diff
@@ -413,6 +420,8 @@ if(CLR_CMAKE_TARGET_UNIX)
         set(CLR_CMAKE_TARGET_UNIX_X86 1)
     elseif(CLR_CMAKE_TARGET_ARCH STREQUAL s390x)
         set(CLR_CMAKE_TARGET_UNIX_S390X 1)
+    elseif(CLR_CMAKE_TARGET_ARCH STREQUAL ppc64le)
+	set(CLR_CMAKE_TARGET_UNIX_POWERPC64 1)
```
Comment (akoeplinger) [inline eng/native/configureplatform.cmake:424]: spaces for indentation:
```suggestion
        set(CLR_CMAKE_TARGET_UNIX_POWERPC64 1)
```

**Comment by Sapana-Khemkar** [comment]
rebased to resolve conflicts

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**